src/Workspaces/Core/Portable/Indentation/AbstractIndentationService.cs

```diff
@@ -30,28 +29,15 @@ private IEnumerable<AbstractFormattingRule> GetFormattingRules(Document document
             return formattingRules;
         }
 
-        public virtual IndentationResult? GetDesiredIndentation(Document document, int lineNumber, CancellationToken cancellationToken)
+        public IndentationResult GetDesiredIndentation(Document document, int lineNumber, CancellationToken cancellationToken)
         {
             var indenter = GetIndenter(document, lineNumber, cancellationToken);
 
             var indentStyle = indenter.OptionSet.GetOption(FormattingOptions.SmartIndent, document.Project.Language);
             if (indentStyle == FormattingOptions.IndentStyle.None)
             {
                 // If there is no indent style, then do nothing.
-                return null;
-            }
-
-            // There are two important cases for indentation.  The first is when we're simply
-            // trying to figure out the appropriate indentation on a blank line (i.e. after
-            // hitting enter at the end of a line, or after moving to a blank line).  The 
-            // second is when we're trying to figure out indentation for a non-blank line
-            // (i.e. after hitting enter in the middle of a line, causing tokens to move to
-            // the next line).  If we're in the latter case, we defer to the Formatting engine
-            // as we need it to use all its rules to determine where the appropriate location is
-            // for the following tokens to go.
```

Comment:
don't need any of this anymore.  the indentation service is now responsible for all of this.

Reply:
hmm.. how?

Reply:
there is so many behavior change, it looks like we get rid of the code by basically not doing stuff we used to do.

Reply:
there should be no behavior changes here.  We got rid of code that wasn't serving an actual purpose.  And this is demonstrable through the tests that show no behavior changes.  The main change is that we simply moved 'smart token indent' under hte main indent codepath, instead of having to happen out-of-band.